### PR TITLE
[Feature] Support show create external database

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this:
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Problem Summary(Required):

```
MySQL > show create database jlt_ice_hdfs;
+--------------+---------------------------------------------------------------------------------------------------------------------------------------+
| Database     | Create Database                                                                                                                       |
+--------------+---------------------------------------------------------------------------------------------------------------------------------------+
| jlt_ice_hdfs | CREATE DATABASE `jlt_ice_hdfs`
PROPERTIES ("location" = "hdfs://xxx") |
+--------------+---------------------------------------------------------------------------------------------------------------------------------------+
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 1]
```diff
@@ -946,11 +946,24 @@ private void handleShowVariables() {
...
         ShowCreateDbStmt showStmt = (ShowCreateDbStmt) stmt;
+        String catalogName = showStmt.getCatalogName();
+        String dbName = showStmt.getDb();
         List<List<String>> rows = Lists.newArrayList();
-        Database db = connectContext.getGlobalStateMgr().getDb(showStmt.getDb());
-        MetaUtils.checkDbNullAndReport(db, showStmt.getDb());
-        rows.add(Lists.newArrayList(showStmt.getDb(),
-                "CREATE DATABASE `" + showStmt.getDb() + "`"));
+
+        if (Strings.isNullOrEmpty(catalogName) || CatalogMgr.isInternalCatalog(catalogName)) {
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:953]: why not get catalog name from session

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:953]: I think they are equivalent.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 14 (57.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|8|14|57.14%|[959, 960, 961, 962, 963, 964]|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 1 above)
Comment (DorianZheng) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:953]: So when is catalog name null?

Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:953]: `catalogName` is unset for internal database

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 8 / 14 (57.14%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|8|14|57.14%|[959, 960, 961, 962, 963, 964]|

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:953]: You need to fill database and catalog in the ShowStmtAnalyzer.visitShowCreateDbStatement()

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java [hunk 2]
```diff
@@ -946,11 +946,24 @@ private void handleShowVariables() {
...
+            Database db = connectContext.getGlobalStateMgr().getDb(dbName);
+            MetaUtils.checkDbNullAndReport(db, showStmt.getDb());
+            rows.add(Lists.newArrayList(showStmt.getDb(),
+                    "CREATE DATABASE `" + showStmt.getDb() + "`"));
+        } else { // external database
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:958]: merge this two block logic. and check if location is null , if it's null, don't display.

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 1 above)
Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java:953]: filled in BasicDbStmtAnalyzer.BasicDbStmtAnalyzerVisitor#visitShowCreateDbStatement

File: fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java (hunk 2 above)
Comment (letian-jiang) [reply fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java]: done

**Review by stephen-shelby — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 8 / 10 (80.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/ShowExecutor.java|8|10|80.00%|[957, 964]|

**Pull request merged**